### ISSUE TITLE
input_select in entities card doesn't refresh UI if modified from script AND same position selected (new option value from a new option list, but at the same position)

### ISSUE THREAD

**Issue by lmagyar** [home-assistant/frontend#13456]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

When an `input_select`'s complete `options` list (and certainly the selected value) is changed from a script, but the new selected value has the same **position** in the new list as the old selected value had in the old list, the UI won't refresh if it is in an `entities` card. Note: when it is in an `entity` card, it works correctly.

My guess is that when it is in an entities card, it tries to detect the change only checking the position of the selected option, not the concrete selected value of the option.

I've spent 2 days to make this issue reproducible. See below.

### Describe the behavior you expected

Both displayed value in the entities card and in the entity card is changed.

### Steps to reproduce the issue

Configuration:

```
input_select:

  test:
    options:
      - "initial"

script:

  input_select_test_works:
    mode: queued
    sequence:
      - variables:
          # just 9 random number as string, first digit goes 1..9
          new_options: '{{ range(now().second * 10 + range(0, 10) | random + 1000, 10000, 1000) | map("string") | list }}'
          #############################################################
          # THIS IS THE ONLY DIFFERENCE, random item, RANDOM POSITION #
          #############################################################
          new_option: '{{ new_options | random }}'
          old_option: '{{ states("input_select.test") }}'
      - service: input_select.set_options
        data:
          entity_id: "input_select.test"
          options: '{{ new_options + [old_option]}}'
      - service: input_select.select_option
        data:
          entity_id: "input_select.test"
          option: '{{ new_option }}'
      - service: input_select.set_options
        data:
          entity_id: "input_select.test"
          options: '{{ new_options }}'

  input_select_test_fails:
    mode: queued
    sequence:
      - variables:
          # just 9 random number as string, first digit goes 1..9
          new_options: '{{ range(now().second * 10 + range(0, 10) | random + 1000, 10000, 1000) | map("string") | list }}'
          #########################################################
          # THIS IS THE ONLY DIFFERENCE, always the SAME POSITION #
          #########################################################
          new_option: '{{ new_options[0] }}'
          old_option: '{{ states("input_select.test") }}'
      - service: input_select.set_options
        data:
          entity_id: "input_select.test"
          options: '{{ new_options + [old_option]}}'
      - service: input_select.select_option
        data:
          entity_id: "input_select.test"
          option: '{{ new_option }}'
      - service: input_select.set_options
        data:
          entity_id: "input_select.test"
          options: '{{ new_options }}'
```

Raw Lovelace config:
```
views:
  - badges: []
    cards:
      - type: entity
        entity: input_select.test
      - type: entities
        entities:
          - entity: input_select.test
      - type: button
        name: Test FAILS
        show_icon: false
        tap_action:
          action: call-service
          service: script.input_select_test_fails
      - type: button
        name: Test WORKS
        show_icon: false
        tap_action:
          action: call-service
          service: script.input_select_test_works
```

---

- when you use the buttons, 9 new random values are generated, like:
  ```
  [
    "1xxx",
    "2xxx",
    "3xxx",
    "4xxx",
    "5xxx",
    "6xxx",
    "7xxx",
    "8xxx",
    "9xxx"
  ]
  ```
- fails: always the first option is selected, the UI never refreshes, you see the same `1xxx` value
- works: random option is selected, you see the changing values like `2xxx`, `6yyy`, `3zzz`, etc.

### What version of Home Assistant Core has the issue?

2022.8.6 (frontend: 20220802.0)

### What was the last working version of Home Assistant Core?

???

### In which browser are you experiencing the issue with?

Google Chrome 104.0.5112.101, FireFox 103.0.2 (64-bit), Companion app

### Which operating system are you using to run this browser?

Windows, Android

**Comment by lmagyar** [comment]
Related: #11991

**Comment by lmagyar** [comment]
Tested with latest 2022.11.4, still fails.

**Comment by lmagyar** [comment]
Thank you for locating the problem! :) My web development knowledge is zero, so I couldn't locate it and I can't test it also. :(

But based on #11991 and the 2 PRs that fixed that problem #12703 and #12181, the hui-_**input**_-select-entity-row.ts and hui-select-entity-row.ts had to be fixed together, and now I see they are differ on this line, so _**MAYBE**_ the correct solution is this (I don't know what I'm talking about, I just see the difference):

```
                (option) =>
                  html`
                    <mwc-list-item .value=${option}
                      >${(stateObj.attributes.device_class &&
                        this.hass!.localize(
                          `component.select.state.${stateObj.attributes.device_class}.${option}`
                        )) ||
                      this.hass!.localize(
                        `component.select.state._.${option}`
                      ) ||
                      option}
                    </mwc-list-item>
                  `
```

So if you could please test this, and if this solves the problem, I _**think**_ this should be a PR.

**Comment by karwosts** [comment]
I found out in some more testing that my original suggestion did not fix some other corner cases related to this issue, so I went with a different fix that fixes all the issues I could find. Submitted PR linked above.
